test/WebJobs.Script.Tests.Integration/ApplicationInsights/ApplicationInsightsEndToEndTestsBase.cs

```diff
@@ -223,19 +223,23 @@ private void ValidateException(ExceptionTelemetry telemetry, string expectedOper
             // Validate the host startup traces. Order by message string as the requests may come in
             // slightly out-of-order or on different threads
             TraceTelemetry[] traces = null;
+            int expectedCount = 13;
 
             await TestHelpers.Await(() =>
             {
                 traces = _fixture.Channel.Telemetries
                     .OfType<TraceTelemetry>()
-                    .Where(t => t.Context.Operation.Id == null)
+                    .Where(t => t.Properties[LogConstants.CategoryNameKey].ToString().StartsWith("Host."))
                     .OrderBy(t => t.Message)
                     .ToArray();
 
-                return traces.Length >= 11;
+                // When these two messages are logged, we know we've completed initialization.
+                return traces
+                .Where(t => t.Message.Contains("Host lock lease acquired by instance ID") || t.Message.Contains("Job host started"))
+                .Count() == 2;
             });
```

Comment:
These counts were all out-of-whack. But checking for the actual logs we want in this `Await` should be more reliable and provide a better error message in the future as it will include all of the startup logs to compare against.